#cloud-config
hostname: {{node.name}}
debug:
   verbose: true
   output: /var/log/cloud-init/
users:
 - default
 - name: stack
   lock-passwd: false
   passwd: $6$rounds=4096$iX2Wgme5F1tgJk$duqxV0FKu7A6zLgcxfVF6s5ipzIVh4q.1yPjOwpFwOS3QcL68rVYXR0ao.3XJFCtd700SaNkDV4KmWvy3LkTb.
   sudo: "ALL=(ALL) NOPASSWD:ALL"
   groups: sudo
   shell: /bin/bash
   ssh-authorized-keys:
    - {{app.key_pair.public}}
{% if app.server_key %}
    - {{app.server_key}}
{% endif %}
bootcmd:
 - echo nameserver {{app.network.gateway}} > /etc/resolv.conf
{% for node in app.nodes %}
 - echo {{node.address}} {{node.name}} >> /etc/hosts
{% endfor %}
packages:
 - git
 - traceroute
 - python
 - python-simplejson

phone_home:
  url: http://{{app.network.gateway}}:{{app.callback_port}}
  post: all
ssh_keys:
  rsa_private: |
     {{app.key_pair.private | indent(5, False)}}
  rsa_public: {{app.key_pair.public}}
write_files:
 - path: /tmp/id_rsa
   content: |
     {{app.key_pair.private | indent(5, False)}}
   permissions: '0600'
runcmd:
  - whoami
  - ls -la /tmp
  - chown stack:stack /home/stack
  - mv /tmp/id_rsa /home/stack/.ssh/id_rsa
  - chown stack:stack /home/stack/.ssh/id_rsa
